plugin/trino-paimon/src/main/java/io/trino/plugin/paimon/PaimonMetadata.java

```diff
@@ -0,0 +1,457 @@
+    {
+        SchemaTableName table = tableMetadata.getTable();
+        Identifier identifier = Identifier.create(table.getSchemaName(), table.getTableName());
+
+        if (identifier.getTableName().length() > MAX_TABLE_LENGTH) {
+            throw new TrinoException(NOT_SUPPORTED, format("Table name must be shorter than or equal to '%s' characters but got '%s'", MAX_TABLE_LENGTH, identifier.getTableName().length()));
+        }
+
+        try {
+            catalog.createTable(session, identifier, prepareSchema(tableMetadata), false);
+        }
+        catch (DatabaseNotExistException e) {
+            throw new TrinoException(SCHEMA_NOT_FOUND, format("Schema %s not found", table.getSchemaName()));
+        }
+        catch (Catalog.TableAlreadyExistException e) {
+            switch (saveMode) {
+                case IGNORE -> {}
+                case REPLACE -> {
+                    try {
+                        catalog.dropTable(session, identifier, false);
+                        catalog.createTable(session, identifier, prepareSchema(tableMetadata), true);
+                    }
+                    catch (DatabaseNotExistException ex) {
+                        throw new RuntimeException(format("database not existed: '%s'", table.getTableName()));
+                    }
+                    catch (Catalog.TableAlreadyExistException ex) {
+                        throw new RuntimeException(format("table already exists: '%s'", table.getTableName()));
+                    }
+                    catch (TableNotExistException ex) {
+                        throw new RuntimeException(format("table not exists: '%s'", table.getTableName()));
+                    }
+                }
+                case FAIL -> throw new RuntimeException(format("table already existed: '%s'", table.getTableName()));
+            }
+        }
+    }
+
+    private static Schema prepareSchema(ConnectorTableMetadata tableMetadata)
+    {
+        Map<String, Object> properties = new HashMap<>(tableMetadata.getProperties());
+        Schema.Builder builder = Schema.newBuilder()
+                .primaryKey(PaimonTableOptions.getPrimaryKeys(properties))
+                .partitionKeys(PaimonTableOptions.getPartitionedKeys(properties))
+                .comment(tableMetadata.getComment().orElse(null));
+
+        for (ColumnMetadata column : tableMetadata.getColumns()) {
+            builder.column(column.getName(), toPaimonType(column.getType()), column.getComment());
+        }
+
+        PaimonTableOptionUtils.buildOptions(builder, properties);
+
+        return builder.build();
+    }
+
+    @Override
+    public void dropTable(ConnectorSession session, ConnectorTableHandle tableHandle)
+    {
+        PaimonTableHandle table = (PaimonTableHandle) tableHandle;
+        try {
+            catalog.dropTable(session, new Identifier(table.getSchemaName(), table.getTableName()), false);
```

Comment:
Supporting drop a table without creation support is little weird. Can we remove this support from this PR?